sdk/identity/identity/src/msal/browserFlows/msalAuthCode.ts

```diff
@@ -3,249 +3,352 @@
+      const msalApp = await getApp();
+      const response = await msalApp.acquireTokenSilent(parameters);
+      return handleResult(scopes, response);
     } catch (err: any) {
       throw handleMsalError(scopes, err, options);
     }
   }
-
   /**
    * Attempts to retrieve the token in the browser.
    */
-  protected async doGetToken(
+  async function doGetToken(
     scopes: string[],
-    options?: CredentialFlowGetTokenOptions,
+    getTokenOptions?: CredentialFlowGetTokenOptions,
   ): Promise<AccessToken> {
-    const account = await this.getActiveAccount();
-    if (!account) {
+    const activeAccount = await getActiveAccount();
+    if (!activeAccount) {
       throw new AuthenticationRequiredError({
         scopes,
-        getTokenOptions: options,
+        getTokenOptions,
         message:
           "Silent authentication failed. We couldn't retrieve an active account from the cache.",
       });
     }
 
     const parameters: msalBrowser.RedirectRequest = {
-      authority: options?.authority || this.msalConfig.auth.authority!,
-      correlationId: options?.correlationId,
-      claims: options?.claims,
-      account: publicToMsal(account),
-      loginHint: this.loginHint,
+      authority: getTokenOptions?.authority || msalConfig.auth.authority!,
+      correlationId: getTokenOptions?.correlationId,
+      claims: getTokenOptions?.claims,
+      account: publicToMsal(activeAccount),
+      loginHint: loginHint,
       scopes,
     };
-    const app = await this.getApp();
-    switch (this.loginStyle) {
+    const msalApp = await getApp();
+    switch (loginStyle) {
       case "redirect":
         // This will go out of the page.
         // Once the InteractiveBrowserCredential is initialized again,
         // we'll load the MSAL account in the constructor.
 
-        await app.acquireTokenRedirect(parameters);
+        await msalApp.acquireTokenRedirect(parameters);
         return { token: "", expiresOnTimestamp: 0, tokenType: "Bearer" };
       case "popup":
-        return this.handleResult(scopes, await app.acquireTokenPopup(parameters));
+        return handleResult(scopes, await app.acquireTokenPopup(parameters));
     }
   }
+
+  /**
+   * Calls to the implementation's doGetToken method.
+   */
+  async function getToken(
+    scopes: string[],
+    getTokenOptions: CredentialFlowGetTokenOptions = {},
+  ): Promise<AccessToken> {
+    const getTokenTenantId =
+      processMultiTenantRequest(tenantId, getTokenOptions, additionallyAllowedTenantIds) ||
+      tenantId;
+
+    if (!getTokenOptions.authority) {
+      getTokenOptions.authority = getAuthority(getTokenTenantId, authorityHost);
+    }
+
+    // We ensure that redirection is handled at this point.
+    await handleRedirect();
+
+    if (!(await getActiveAccount()) && !disableAutomaticAuthentication) {
+      await login(scopes);
+    }
+    return getTokenSilent(scopes, getTokenOptions).catch((err) => {
```

Comment:
I try to avoid mixing `async/await` syntax with promise chaining where possible. I know this is mostly existing code so feel free to ignore but it makes things a little harder to reason about.

You could easily make this `async/await` but feel free to do that in a separate PR

Reply:
I have separated this promise chaining with try/catch block so that the logic is easier to parse. Thank you for pointing this out 👍